File: workflow/plugins/python/list/list_length/list_length.py
"""Workflow plugin: get list length."""

from ...base import NodeExecutor


class ListLength(NodeExecutor):
    """Get list length."""

    node_type = "list.length"
    category = "list"
    description = "Get list length"

    def execute(self, inputs, runtime=None):
        array = inputs.get("array", inputs.get("list", []))
        return {"result": len(array) if array is not None else 0}

File: workflow/plugins/go/list/list_length/list_length.go
// Package list_length provides a workflow plugin for getting list length.
package list_length

// ListLength implements the NodeExecutor interface for getting list length.
type ListLength struct {
	NodeType    string
	Category    string
	Description string
}

// NewListLength creates a new ListLength instance.
func NewListLength() *ListLength {
	return &ListLength{
		NodeType:    "list.length",
		Category:    "list",
		Description: "Get the length of a list",
	}
}

// Execute runs the plugin logic.
func (p *ListLength) Execute(inputs map[string]interface{}, runtime interface{}) map[string]interface{} {
	list, ok := inputs["list"].([]interface{})
	if !ok {
		return map[string]interface{}{"result": 0}
	}

	return map[string]interface{}{"result": len(list)}
}
